### PR TITLE
doc: add hugegraph-ai doc

### PR DESCRIPTION
**Pull request by simon824**
Link https://github.com/apache/incubator-hugegraph-ai/pull/30

### CODE REVIEW

**Review by liuxiaocs7 — COMMENTED** [review]
when exec rag, seems download nltk stopwords fails:

image


image

File: content/cn/docs/quickstart/hugegraph-ai.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+### 4 使用说明
+#### 4.1 通过 LLM 在 HugeGraph 中构建知识图谱
+##### 4.1.1 通过 gradio 交互式界面构建知识图谱
+- 参数说明：
+  - Text: 输入的文本。 
+  - Schema：接受以下两种类型的文本： 
+    - 用户定义的 JSON 格式模式。 
+    - 指定 HugeGraph 图实例的名称，它将自动提取图的模式。
+  - Disambiguate word sense：是否进行词义消除歧义。 
+  - Commit to hugegraph：是否将构建的知识图提交到 HugeGraph 服务器
```
Comment (liuxiaocs7) [inline content/cn/docs/quickstart/hugegraph-ai.md:37]:
```suggestion
  - Commit to hugegraph：是否将构建的知识图谱提交到 HugeGraph 服务器
```

File: content/cn/docs/quickstart/hugegraph-ai.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    .disambiguate_word_sense().print_result()
+    .commit_to_hugegraph()
+    .run()
+)
+```
+- 时序图
+![gradio配置](/docs/images/kg-uml.png)
+
+
+1. 初始化: 初始化 LLMs 实例，获取 LLM 模型，然后创建图谱构建的任务实例 `KgBuilder`，KgBuilder 中定义了多个 operator ，用户可以根据需求自由组合达到目的 。（tip: `print_result()` 可以在控制台打印每一步输出的结果，不影响整体执行逻辑）
```
Comment (liuxiaocs7) [inline content/cn/docs/quickstart/hugegraph-ai.md:62]:
```suggestion
1. 初始化: 初始化 LLMs 实例，获取 LLM，然后创建图谱构建的任务实例 `KgBuilder`，KgBuilder 中定义了多个 operator，用户可以根据需求自由组合达到目的 。（tip: `print_result()` 可以在控制台打印每一步输出的结果，不影响整体执行逻辑）
```

File: content/cn/docs/quickstart/hugegraph-ai.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    - 从用户定义的模式导入，接受用户定义的 JSON 格式模式。
+    - 从提取结果导入（即将发布）
+```python
+# Import schema from a HugeGraph instance
+builder.import_schema(from_hugegraph="test_graph").print_result()
+# Import schema from an extraction result
+builder.import_schema(from_extraction="xxx").print_result()
+# Import schema from user-defined schema
+builder.import_schema(from_user_defined="xxx").print_result()
+```
```
Comment (liuxiaocs7) [inline content/cn/docs/quickstart/hugegraph-ai.md:79]: Is it better to keep the description and code in the same order?

**Comment by simon824** [comment]
> when exec rag, seems download nltk stopwords fails:
> 
> image
> 
> image

Did you try again several times? It might be a network issue.

**Comment by liuxiaocs7** [comment]
> Did you try again several times? It might be a network issue.

Yes, I have executed it manually many times and the result is like this, log:


```
  File "/home/root1/software/miniconda3/envs/hg-ai/lib/python3.11/site-packages/nltk/data.py", line 583, in find
    raise LookupError(resource_not_found)
LookupError: 
**********************************************************************
  Resource stopwords not found.
  Please use the NLTK Downloader to obtain the resource:

  >>> import nltk
  >>> nltk.download('stopwords')
  
  For more information see: https://www.nltk.org/data.html

  Attempted to load corpora/stopwords

  Searched in:
    - '/home/root1/nltk_data'
    - '/home/root1/software/miniconda3/envs/hg-ai/nltk_data'
    - '/home/root1/software/miniconda3/envs/hg-ai/share/nltk_data'
    - '/home/root1/software/miniconda3/envs/hg-ai/lib/nltk_data'
    - '/usr/share/nltk_data'
    - '/usr/local/share/nltk_data'
    - '/usr/lib/nltk_data'
    - '/usr/local/lib/nltk_data'
    - '/tmp/hugegraph_llm'
**********************************************************************

```

File: content/cn/docs/quickstart/hugegraph-ai.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+export PYTHONPATH=${PROJECT_ROOT_DIR}/hugegraph-llm/src:${PROJECT_ROOT_DIR}/hugegraph-python-client/src
+python3 ./hugegraph-llm/src/hugegraph_llm/utils/gradio_demo.py
+```
+- 配置 HugeGraph 数据库连接信息和 LLM 模型信息，可以通过两种方式配置：
+  1. 配置 `./hugegraph-llm/src/config/config.ini` 文件
+  2. 在 gradio 中配置，如图所示:
+  ![gradio配置](/docs/images/gradio-config.png)
+- 离线下载 NLTK stopwords 
+```bash
+python3 ./hugegraph_llm/operators/common_op/nltk_helper.py
```
Comment (simon824) [inline content/cn/docs/quickstart/hugegraph-ai.md:28]: @liuxiaocs7 We can download stopwords offline by running the command below
```python
python3 ./hugegraph_llm/operators/common_op/nltk_helper.py
```

**Comment by liuxiaocs7** [comment]
Using the proxy in the command line can download normally, but encountered new problems when executing RAG.

1. First init HG

image

2. RAG

image


```bash
INFO:     192.168.110.32:11406 - "POST /queue/join HTTP/1.1" 200 OK
INFO:     192.168.110.32:11406 - "GET /queue/data?session_hash=804xxsr8w7f HTTP/1.1" 200 OK
KEYWORDS: ['Al Pacino', 'tell me about', 'Hollywood', 'film', 'Al', 'Pacino', 'tell', 'actor']
Traceback (most recent call last):
  File "/home/root1/software/miniconda3/envs/hg-ai/lib/python3.11/site-packages/gradio/queueing.py", line 495, in call_prediction
    output = await route_utils.call_process_api(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/root1/software/miniconda3/envs/hg-ai/lib/python3.11/site-packages/gradio/route_utils.py", line 235, in call_process_api
    output = await app.get_blocks().process_api(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/root1/software/miniconda3/envs/hg-ai/lib/python3.11/site-packages/gradio/blocks.py", line 1627, in process_api
    result = await self.call_function(
             ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/root1/software/miniconda3/envs/hg-ai/lib/python3.11/site-packages/gradio/blocks.py", line 1173, in call_function
    prediction = await anyio.to_thread.run_sync(
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/root1/software/miniconda3/envs/hg-ai/lib/python3.11/site-packages/anyio/to_thread.py", line 56, in run_sync
    return await get_async_backend().run_sync_in_worker_thread(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/root1/software/miniconda3/envs/hg-ai/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 2144, in run_sync_in_worker_thread
    return await future
           ^^^^^^^^^^^^
  File "/home/root1/software/miniconda3/envs/hg-ai/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 851, in run
    result = context.run(func, *args)
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/root1/software/miniconda3/envs/hg-ai/lib/python3.11/site-packages/gradio/utils.py", line 690, in wrapper
    response = f(*args, **kwargs)
               ^^^^^^^^^^^^^^^^^^
  File "/home/root1/lx/incubator-hugegraph-ai/./hugegraph-llm/src/hugegraph_llm/utils/gradio_demo.py", line 83, in graph_rag
    .run(verbose=True)
     ^^^^^^^^^^^^^^^^^
  File "/home/root1/lx/incubator-hugegraph-ai/hugegraph-llm/src/hugegraph_llm/operators/graph_rag_task.py", line 90, in run
    context = op.run(context)
              ^^^^^^^^^^^^^^^
  File "/home/root1/lx/incubator-hugegraph-ai/hugegraph-llm/src/hugegraph_llm/operators/hugegraph_op/graph_rag_query.py", line 130, in run
    raise RuntimeError("Unsupported ID format for Graph RAG.")
RuntimeError: Unsupported ID format for Graph RAG.

```

**Comment by liuxiaocs7** [comment]
Hi, @imbajin, any suggestions?

**Comment by simon824** [comment]
> after clean data and then re-init graph got the result.
> 
> ```
> curl -v -X DELETE 192.168.110.215:8080/graphs/hugegraph/clear?confirm_message=I%27m+sure+to+delete+all+data
> ```

Added `clear_graph_all_data()` in https://github.com/apache/incubator-hugegraph-ai/pull/30/commits/c5a5a47bbbf8e10fcc04b4404d05fbfbd12e5227

**Review by liuxiaocs7 — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -50,6 +50,7 @@ The functions of this system include but are not limited to:
...
 - [HugeGraph-Hubble](https://hugegraph.apache.org/docs/quickstart/hugegraph-hubble): HugeGraph-Hubble is HugeGraph's web visualization management platform, a one-stop visual analysis platform. The platform covers the whole process from data modeling, to rapid data import, to online and offline analysis of data, and unified management of graphs;
 - [HugeGraph-Tools](https://hugegraph.apache.org/docs/quickstart/hugegraph-tools): HugeGraph-Tools is HugeGraph's deployment and management tools, including functions such as managing graphs, backup/restore, Gremlin execution, etc.
+- [HugeGraph-Ai (Beta)](https://hugegraph.apache.org/docs/quickstart/hugegraph-ai): HugeGraph-Ai is a tool that integrates HugeGraph and artificial intelligence (AI), including applications combined with large models, integration with graph machine learning components, etc., to provide comprehensive support for developers to use HugeGraph's AI capabilities in projects.
```
Comment (imbajin) [inline README.md:53]: could also add to other repo's README now? (for release 1.3)

**Review by imbajin — APPROVED** [review]

**Pull request merged**